pandas/tests/test_reshape.py

```diff
@@ -171,6 +172,33 @@ def test_basic(self):
         expected.index = list('ABC')
         assert_frame_equal(get_dummies(s_series_index, sparse=self.sparse), expected)
 
+    def test_basic_types(self):
+        # GH 10531
+        s_list = list('abc')
+        s_series = Series(s_list)
+        s_df = DataFrame({'a': [0, 1, 0, 1, 2],
+                          'b': ['A', 'A', 'B', 'C', 'C'],
+                          'c': [2, 3, 3, 3, 2]})
+
+        if not self.sparse:
+            exp_df_type = DataFrame 
+            exp_blk_type = pd.core.internals.FloatBlock
+        else:
+            exp_df_type = SparseDataFrame
+            exp_blk_type = pd.core.internals.SparseBlock
+
+        self.assertEqual(type(get_dummies(s_list, sparse=self.sparse)), exp_df_type)
+        self.assertEqual(type(get_dummies(s_series, sparse=self.sparse)), exp_df_type)
+
+        r = get_dummies(s_df, sparse=self.sparse, columns=s_df.columns)
+        self.assertEqual(type(r), exp_df_type)
+
+        r = get_dummies(s_df, sparse=self.sparse, columns=['a'])
+        self.assertEqual(type(r[['a_0']]._data.blocks[0]), exp_blk_type)
+        self.assertEqual(type(r[['a_1']]._data.blocks[0]), exp_blk_type)
+        self.assertEqual(type(r[['a_2']]._data.blocks[0]), exp_blk_type)
+
```

Comment:
ok looks reasonable

ping when green

FYI - don't take lines out of the what's new I put them there on purpose - helps avoid merge conflicts (in fact u have one now because of that)